scripts/claimLineaL2Message.ts

```diff
@@ -0,0 +1,123 @@
+
+/**
+ * Script to claim L1->L2 messages sent by Linea_Adapter. Run via
+ * ```
+ * yarn hardhat run ./scripts/claimLineaL2Message.ts --network linea-goerli
+ * ```
+ */
+async function main() {
+  const blockLookback = parseInt(process.env.BLOCK_LOOKBACK || "8640");
+
+  const l2ChainId = parseInt(await hre.getChainId());
+  const l1ChainId = parseInt(await hre.companionNetworks.l1.getChainId());
+  const l1Provider = new ethers.providers.JsonRpcProvider(
+    getNodeUrl(l1ChainId === 1 ? "mainnet" : "goerli", true, l1ChainId)
+  );
+  const l1Signer = ethers.Wallet.fromMnemonic((hre.network.config.accounts as any).mnemonic).connect(l1Provider);
+
+  const sdk = new LineaSDK({
+    l1RpcUrl: getNodeUrl(l1ChainId === 1 ? "mainnet" : "goerli", true, l1ChainId),
+    l2RpcUrl: `https://${l2ChainId === 59144 ? "linea" : "linea-goerli"}.infura.io/v3/${process.env.INFURA_API_KEY}`,
+    l1SignerPrivateKey: l1Signer.privateKey,
+    l2SignerPrivateKey: l1Signer.privateKey,
+    network: l1ChainId === 1 ? "linea-mainnet" : "linea-goerli",
+    mode: "read-write",
+  });
+  const l1MessageService = sdk.getL1Contract(L1_ADDRESS_MAP[l1ChainId].lineaMessageService);
+  const l2MessageService = sdk.getL2Contract(L2_ADDRESS_MAP[l2ChainId].lineaMessageService);
+
+  const hubPoolDeployment = await hre.companionNetworks.l1.deployments.get("HubPool");
+  const lineaAdapter = (await getContractFactory("Linea_Adapter", { signer: l1Signer })).attach(
+    hubPoolDeployment.address
+  );
+
+  console.log("\nL1 chain ID:", l1ChainId);
+  console.log("L2 chain ID:", l2ChainId);
+  console.log("Signer address:", l1Signer.address);
+
+  console.log("\nQuerying 'MessageRelayed' or 'TokensRelayed' events from HubPool:", lineaAdapter.address);
+
+  const l1LatestBlock = await l1Provider.getBlockNumber();
+  const l1InitialFromBlock = l1LatestBlock - blockLookback;
+  let l1FromBlock = l1InitialFromBlock;
+
+  const relevantMessageSentEvents = [];
+
+  while (l1FromBlock < l1LatestBlock) {
+    const numBlocks = Math.max(l1LatestBlock - l1FromBlock, 0);
+
+    if (numBlocks === 0) {
+      break;
+    }
+
+    const toBlock = numBlocks > MAX_L1_BLOCK_LOOKBACK ? l1FromBlock + MAX_L1_BLOCK_LOOKBACK : l1FromBlock + numBlocks;
+    console.log(`Querying blocks ${l1FromBlock} - ${toBlock}...`);
+    const [tokensRelayedEvents, messageRelayedEvent] = await Promise.all([
+      lineaAdapter.queryFilter("TokensRelayed", l1FromBlock, toBlock),
+      lineaAdapter.queryFilter("MessageRelayed", l1FromBlock, toBlock),
+    ]);
+    const events = [...tokensRelayedEvents, ...messageRelayedEvent];
+    console.log(
+      `Found ${events.length} events: ${tokensRelayedEvents.length} 'TokensRelayed', ${messageRelayedEvent.length} 'MessageRelayed'`
+    );
+
+    const uniqueTxHashes = new Set(events.map((event) => event.transactionHash));
+    for (const txHash of uniqueTxHashes) {
+      const messageSentEvents = await l1MessageService.getMessagesByTransactionHash(txHash);
```

Comment:
Does this handle well sending multiple messages from the same txHash? In the `finalizer` we've sometimes had issues using the native SDK for a chain to handle the multiple message case so we implemented a helper function `getUniqueLogIndex` [here in use in the Arbitrum finalizer](https://github.com/across-protocol/relayer-v2/blob/master/src/finalizer/utils/arbitrum.ts#L135) to handle this case

Reply:
Good question 🤔 I will take a look at the source code